src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffolderBuilderExtensions.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Scaffolding.Helpers.Services;
+using Microsoft.DotNet.Scaffolding.TextTemplating;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.Blazor.BlazorCrud;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.MinimalApi;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Helpers;
+
+namespace Microsoft.DotNet.Scaffolding.Core.Builder;
+
+internal static class ScaffolderBuilderExtensions
+{
+    public static IScaffoldBuilder WithMinimalApiTextTemplatingStep(this IScaffoldBuilder builder)
+    {
+        var allT4Templates = new TemplateFoldersUtilities().GetAllT4Templates(["MinimalApi"]);
+        string? t4TemplatePath = null;
```

Comment:
Do these lines need to be outside the WithStep call? Thinking about it, calling GetAllT4Templates now would enumerate the files every time the tool ran, even when just running `get-commands` or whatever. If it was inside the WithStep call, it'd be deferred until execution.

Reply:
moved these inside the step since only one `WithStep` is needed.